symbulate/results.py

```diff
@@ -285,23 +288,54 @@ def truncate(result):
 class RVResults(Results):
 
     def plot(self, type=None, alpha=None, normalize=True, jitter=False, 
-        bins=30, **kwargs):
+        bins=None, **kwargs):
+        if type is not None:
+            if isinstance(type, str):
+                type = (type,)
+            elif not isinstance(type, (tuple, list)):
+                raise Exception("I don't know how to plot a " + str(type))
         
         dim = get_dimension(self)
         if dim == 1:
             counts = self._get_counts()
+            heights = counts.values()
+            discrete = discrete_check(heights)
             if type is None:
-                heights = counts.values()
-                if sum([(i > 1) for i in heights]) > .8 * len(heights):
-                    type = "impulse"
+                if discrete:
+                    type = ("impulse",)
                 else:
-                    type = "bar"
-            if type == "bar":
-                if alpha is None:
-                    alpha = .5
-                plt.hist(self, normed=normalize, alpha=alpha, bins=bins, **kwargs)
+                    type = ("hist",)
+            if alpha is None:
+                alpha = .5
+            if bins is None:
+                bins = 30
+
+            fig = plt.gcf()
+            ax = plt.gca()
+            color = get_next_color(ax)
+            
+            if 'density' in type:
+                if discrete:
+                    xs = sorted(list(counts.keys()))
+                    ys = []
+                    for val in xs:
+                        ys.append(counts[val] / len(self))
+                    ax.plot(xs, ys, marker='o', color=color, linestyle='-')
+                    if len(type) == 1:
+                        plt.ylabel('Relative Frequency')
+                else:
+                    density = gaussian_kde(self)
+                    density.covariance_factor = lambda: 0.25
+                    density._compute_covariance()
```

Comment:
The logic in this block of code appears multiple times in this file (again for 2 variables). Try factoring out into plot.py?